operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/dependent/kubernetes/GenericKubernetesResourceMatcher.java

```diff
@@ -24,6 +23,9 @@ public class GenericKubernetesResourceMatcher<R extends HasMetadata, P extends H
   private static final String OP = "op";
   public static final String METADATA_LABELS = "/metadata/labels";
   public static final String METADATA_ANNOTATIONS = "/metadata/annotations";
+  // without knowing the CRD we cannot ignore status as it may not be a subresource, so if it's
+  // included we expect it to match
+  private static Set<String> NOT_OTHER_FIELDS = Set.of(SPEC, "/metadata", "/apiVersion", "/kind");
```

Comment:
Why SPEC is here? we want to match SPEC, or? I'm reading it wrong

Reply:
> spec and metadata have their own handling, so that's not covered by the "other fields" matching.

sorry totally missed that.

Reply:
Yet another tweak would be to change the logic to just walk through the diffs once and do prefix matching to determine if the change is allowable.

Reply:
> You would only want to do that if you suspect that users are including non-matching status that is not meaningful on their dependent resources. My guess would be that they are not including status, and if they are it would be because it's coming from the actual state (and would match), or that it's not a subresource and needs to be considered in matching. But if you feel strongly about it, it can be excluded.

So these are resources which usually created, but not managed by the controller, so in normal case we apply the desired state using SSA, that is usually the spec. We don't care about the status at all at this phase, we might case in a postCondition, or when we are setting the status for the primary resource.

Reply:
>  We don't care about the status at all at this phase,

If you assume it's a subresource.  If not, then you do.

Reply:
yep, true, but again that should not be a feature flag? we are comparing the desired with the actual, the actual would contain the status even if it is a subresource, but the desired won't, wouldn't this always fail to match in that case?

Reply:
> we are comparing the desired with the actual, the actual would contain the status even if it is a subresource, but the desired won't, wouldn't this always fail to match in that case?

Ah, the issue you are raising is with the true parameter with https://github.com/operator-framework/java-operator-sdk/pull/2014/files#diff-3425845ceca0cbb05292b87cdceec9960b4342cefac82c18ab929a5ef22e5288R211 - if someone leaves the status unpopulated in the desired (which is expected) the exact match here will cause a problem.  I must have been implicitly assuming that all are add ops would apply here, but that would obviously require a separate case.  Let me make another revision and add a test.

Reply:
Further refactored to clarify the matching algorithm and added status as excluded.